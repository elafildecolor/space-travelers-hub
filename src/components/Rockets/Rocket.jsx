import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import Carousel from 'react-bootstrap/Carousel';
import Figure from 'react-bootstrap/Figure';

const Rocket = ({
  rocketName,
  description,
  flickrImages,
  isReserved,
  handleClick,
}) => (
  <Container>
    <Row>
      <Col xs={12} lg={5} xl={4}>
        <Carousel>
          {
            flickrImages.map(({ imgSrc, imgId }) => (
              <Carousel.Item key={imgId}>
                <Figure>
                  <Figure.Image
                    alt="171x180"
                    src={imgSrc}
                  />
                </Figure>
              </Carousel.Item>
            ))
          }
        </Carousel>
      </Col>
      <Col xs={12} lg={7} xl={8}>
        <Card style={{ border: 0 }}>
          <Card.Body style={{ padding: 0 }}>
            <Card.Title>{rocketName}</Card.Title>
            <Card.Text>{description}</Card.Text>
            <Button
              variant={isReserved ? 'outline-secondary' : 'primary'}
              onClick={handleClick}
            >
              {isReserved ? 'Cancel Reservation' : 'Reserve Rocket'}
            </Button>
          </Card.Body>
        </Card>
      </Col>
    </Row>
  </Container>
);

Rocket.propTypes = {
  rocketName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickrImages: PropTypes.arrayOf(PropTypes.shape({
    imgSrc: PropTypes.string.isRequired,
    imgId: PropTypes.number.isRequired,
  })).isRequired,
  isReserved: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Rocket;
