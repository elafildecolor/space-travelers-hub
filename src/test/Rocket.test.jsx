/* eslint-disable */
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Rocket from '../components/Rockets/Rocket';

describe('Rocket', () => {
  let props;

  beforeEach(() => {
    props = {
      id: 1,
      rocketName: 'Halcon 1',
      description: 'A three-stage-to-orbit medium lift launch vehicle',
      flickrImages: [{imgId: 0, imgSrc: 'https://farm1.staticflickr.com/293/32312415025_6841e30bf1_b.jpg'}],
      isReserved: false,
      handleClick: jest.fn(),
    };
  });

  test('Render correctly with all props', () => {
    const tree = render(<Rocket {...props} />);
    expect(tree).toMatchSnapshot();
  });

  test('Render the correct rocket name', () => {
    render(<Rocket {...props} />);
    const element = screen.getByText('Halcon 1');
    expect(element).toBeInTheDocument();
  });

  test('should render the correct button text when reserved is false', () => {
    render(<Rocket {...props} />);
    const element = screen.getByText('Reserve Rocket');
    expect(element).toBeInTheDocument();
  });

  test('should render the correct button text when reserved is true', () => {
    props.isReserved = true;
    render(<Rocket {...props} />);

    const element = screen.getByText('Cancel Reservation');
    expect(element).toBeInTheDocument();
  });

  test('should call handleRocket when button is clicked', () => {
    render(<Rocket {...props} />);
    const element = screen.getByText('Reserve Rocket');
    fireEvent.click(element);
    expect(props.handleClick).toHaveBeenCalledTimes(1);
  });
  
});
