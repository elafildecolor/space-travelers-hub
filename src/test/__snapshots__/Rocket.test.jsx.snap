// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Render correctly with all props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-xl-4 col-lg-5 col-12"
          >
            <div
              class="carousel slide"
            >
              <div
                class="carousel-indicators"
              >
                <button
                  aria-current="true"
                  aria-label="Slide 1"
                  class="active"
                  data-bs-target=""
                  type="button"
                />
              </div>
              <div
                class="carousel-inner"
              >
                <div
                  class="active carousel-item"
                >
                  <figure
                    class="figure"
                  >
                    <img
                      alt="171x180"
                      class="figure-img img-fluid"
                      src="https://farm1.staticflickr.com/293/32312415025_6841e30bf1_b.jpg"
                    />
                  </figure>
                </div>
              </div>
              <a
                class="carousel-control-prev"
                href="#"
                role="button"
                tabindex="0"
              >
                <span
                  aria-hidden="true"
                  class="carousel-control-prev-icon"
                />
                <span
                  class="visually-hidden"
                >
                  Previous
                </span>
              </a>
              <a
                class="carousel-control-next"
                href="#"
                role="button"
                tabindex="0"
              >
                <span
                  aria-hidden="true"
                  class="carousel-control-next-icon"
                />
                <span
                  class="visually-hidden"
                >
                  Next
                </span>
              </a>
            </div>
          </div>
          <div
            class="col-xl-8 col-lg-7 col-12"
          >
            <div
              class="card"
              style="border: 0px;"
            >
              <div
                class="card-body"
                style="padding: 0px;"
              >
                <div
                  class="card-title h5"
                >
                  Halcon 1
                </div>
                <p
                  class="card-text"
                >
                  A three-stage-to-orbit medium lift launch vehicle
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Reserve Rocket
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-xl-4 col-lg-5 col-12"
        >
          <div
            class="carousel slide"
          >
            <div
              class="carousel-indicators"
            >
              <button
                aria-current="true"
                aria-label="Slide 1"
                class="active"
                data-bs-target=""
                type="button"
              />
            </div>
            <div
              class="carousel-inner"
            >
              <div
                class="active carousel-item"
              >
                <figure
                  class="figure"
                >
                  <img
                    alt="171x180"
                    class="figure-img img-fluid"
                    src="https://farm1.staticflickr.com/293/32312415025_6841e30bf1_b.jpg"
                  />
                </figure>
              </div>
            </div>
            <a
              class="carousel-control-prev"
              href="#"
              role="button"
              tabindex="0"
            >
              <span
                aria-hidden="true"
                class="carousel-control-prev-icon"
              />
              <span
                class="visually-hidden"
              >
                Previous
              </span>
            </a>
            <a
              class="carousel-control-next"
              href="#"
              role="button"
              tabindex="0"
            >
              <span
                aria-hidden="true"
                class="carousel-control-next-icon"
              />
              <span
                class="visually-hidden"
              >
                Next
              </span>
            </a>
          </div>
        </div>
        <div
          class="col-xl-8 col-lg-7 col-12"
        >
          <div
            class="card"
            style="border: 0px;"
          >
            <div
              class="card-body"
              style="padding: 0px;"
            >
              <div
                class="card-title h5"
              >
                Halcon 1
              </div>
              <p
                class="card-text"
              >
                A three-stage-to-orbit medium lift launch vehicle
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
